File: qna/src/main/java/com/example/rag/qna/adapter/output/opinion/OpinionPersistencePort.java
package com.example.rag.qna.adapter.output.opinion;

import com.example.rag.qna.domain.Opinion;

import java.util.List;

public interface OpinionPersistencePort {
    void saveOpinion(Opinion opinion);
    void updateOpinion(Opinion opinion);
    void deleteOpinion(String opinionId); // MongoDB 식별자의 문자열 타입 고려
    Opinion findOpinionById(String opinionId); // MongoDB 식별자의 문자열 타입 고려
    List<Opinion> findOpinionsByQuestionId(String questionId, int offset, int limit);
}

File: qna/src/main/kotlin/com/example/rag/qna/adapter/output/opinion/OpinionPersistencePort.kt
package com.example.rag.qna.adapter.output.opinion

import com.example.rag.qna.domain.Opinion

interface OpinionPersistencePort {
    fun saveOpinion(domain: Opinion): OpinionEntity
    fun updateOpinion(domain: Opinion)
    fun deleteOpinion(opinionId: String) // MongoDB 식별자의 문자열 타입 고려
    fun findOpinionById(opinionId: String): Opinion // MongoDB 식별자의 문자열 타입 고려
    fun findOpinionsByQuestionId(questionId: String, offset: Int, limit: Int): List<Opinion>
}